tomli/_re.py

```diff
@@ -12,8 +12,26 @@
 # - 00:32:00
 _TIME_RE_STR = r"([01][0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])(?:\.([0-9]{1,6})[0-9]*)?"
 
-RE_NUMBER = re.compile(
-    r"""
+
+@lru_cache(maxsize=None)
+def regex(name: str) -> re.Pattern:
+    if name == "datetime":
+        return re.compile(
+            fr"""
+([0-9]{{4}})-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])  # date, e.g. 1988-10-27
+(?:
+    [T ]
+    {_TIME_RE_STR}
+    (?:(Z)|([+-])([01][0-9]|2[0-3]):([0-5][0-9]))?     # optional time offset
```

Comment:
I was playing around with the new tests from https://github.com/BurntSushi/toml-test/blob/master/tests/valid/datetime/datetime.toml and noticed that the [RFC](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6) states the lower case "t" and "z" should be allowed here.

If you're interested in syncing those new tests, I updated `burntsushi.py` with that new format. You can copy it from here https://github.com/oittaa/tomli/commit/beed67277e79f0de3d3724e06e9868d19e928065 or I could make a pull request if you'd prefer that?

Reply:
Good catch, thanks! I'd appreciate a lot if you made a PR that fixes the regexes. That way you also get credit for the commit/PR.

The test changes may be a bit tedious for me to review, so I may prefer updating burntsushi tests myself, or at least having them in a separate PR.